src/math/tests/BarycenterTest.cpp

```diff
@@ -237,6 +237,74 @@ BOOST_AUTO_TEST_CASE(BarycenterTriangle2D)
   }
 }
 
+BOOST_AUTO_TEST_CASE(BarycenterTetrahedron)
```

Comment:
Please break this big test into smaller individual tests.

Create a test suite, called `BarycenterTetrahedron` and individual tests for these cases.

You can define common coordinates in [a fixture](https://www.boost.org/doc/libs/1_71_0/libs/test/doc/html/boost_test/tests_organization/fixtures/case.html#boost_test.tests_organization.fixtures.case.example_descr)  and reuse it in the tests.

Reply:
By new test suite you mean a new file? Because the pattern here seems to be "one test file per source file" and that would break it.

Reply:
You can group your tests in a test suite.

```cpp
BOOST_AUTO_TEST_SUITE(Tetrahedron)

// put your tests here

BOOST_AUTO_TEST_SUITE_END()
```